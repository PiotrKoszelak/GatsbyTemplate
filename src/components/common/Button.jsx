import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledButton = styled.button`
  height: 20px;
  background-color: ${({
    theme
  }) => `${theme.colors.green}`};
  color: ${({
    theme
  }) => `${theme.colors.white}`};
  text-align: center;
  width: ${({
    width
  }) => `${width}px`};
  cursor: pointer;
  border-radius: 5px;
  border: none;
`;

const Button = ({
  text, onClick, width
}) => (
  <StyledButton width={width} onClick={onClick}>
    {text}
  </StyledButton>
);

Button.defaultProps = {
  text: 'no text',
  onClick: () => {},
  width: '75px'
};

Button.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func,
  width: PropTypes.string
};

export default Button;
