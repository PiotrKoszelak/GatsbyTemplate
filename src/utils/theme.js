import colors from './colors';

export const theme = {
    colors,
    media: {
        tablet: '@media (max-width: 1100px)',
        phone: '@media (max-width: 600px)'
    }
}

export default theme