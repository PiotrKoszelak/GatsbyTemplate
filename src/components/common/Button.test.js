import React from 'react';
import renderer from 'react-test-renderer';
import {
  ThemeProvider
} from 'styled-components';
import {
  render, screen
} from '@testing-library/react';
import theme from '../../utils/theme';
import Button from './Button';

const ButtonWrapper = ({
  // eslint-disable-next-line react/prop-types
  children
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

describe('Button defaul behavior', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <ButtonWrapper>
          <Button text='Default text' />
        </ButtonWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('displays the correct default title', () => {
    render(
      <ButtonWrapper>
        <Button />
      </ButtonWrapper>
    );
    expect(screen.getByText('no text')).toBeInTheDocument();
  });
});
