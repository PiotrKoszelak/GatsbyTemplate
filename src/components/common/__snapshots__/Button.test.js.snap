// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button defaul behavior renders correctly 1`] = `
<button
  className="sc-AxjAm gqoGzI"
  onClick={[Function]}
  width="75px"
>
  Default text
</button>
`;
